### PR TITLE
Package

### PR DESCRIPTION
**Pull request by meilame-tayebjee**
- Fixed issues when no categorical variables
- Completed docs
- Optional dependencies with poetry.extras + smooth imports when necessary + removed calls to numpy when there is an equivalent in torch. "pytorch_model.py" now does not need numpy.
- Added build_data_loaders (is useful for training externally, e.g. codif APE)
- Debugged predict and predict_and_explain + extended the notebook experiments to include full pipeline until explainability

--> Total independance from numpy might be possible but requires more work ; and might be not relevant anymore (the pytorch model can be distributed without numpy, this is the most important - only optional features are now Numpy-dependent)
--> Ready to release v0.0.0 on PyPI ?

### CODE REVIEW

**Review by micedre — APPROVED** [review]
Great, some minor modification on README, and good to publish for me

File: README.md [hunk 1]
```diff
@@ -1,15 +1,88 @@
-# torch-FastText: Efficient text classification with PyTorch
+## torchFastText
```
Comment (micedre) [inline README.md:1]: Why change the title ?

File: pyproject.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "torchFastText"
-version = "0.0.5"
+version = "0.0.0"
```
Comment (micedre) [inline pyproject.toml:3]: should be 0.0.1 I think

File: README.md (hunk 1 above)
Comment (meilame-tayebjee) [reply README.md:1]: no specific reason ; just feels better to me without "-" (very subjective).

File: pyproject.toml [hunk 3]
```diff
@@ -23,16 +23,16 @@ classifiers = [
...
 numpy = "1.26.4"
-captum = "^0.7.0"
 pytorch-lightning = "^2.4.0"
-unidecode = "^1.3.8"
-nltk = "^3.9.1"
-scikit-learn = "^1.5.2"
+
+[tool.poetry.extras]
+explainability = ["unidecode", "nltk", "captum"]
+preprocess =     ["unidecode", "nltk"]
```
[the remark this answers is not in the document]
Comment (meilame-tayebjee) [reply pyproject.toml:35]: will do

File: pyproject.toml (hunk 2 above)
Comment (meilame-tayebjee) [reply pyproject.toml:3]: ok

File: README.md (hunk 1 above)
Comment (micedre) [reply README.md:1]: I was just talking about the text, I think we should keep the level (and the mini description)  : 

```suggestion
# torchFastText : Efficient text classification with PyTorch
```

**Pull request merged**